tests/tests.py

```diff
@@ -296,40 +300,101 @@ def setUp(self):
         for i in range(0, total):
             result = _create_issue_result_json(i, "summary %s" % i, key="KEY-%s" % i)
             expected_results.append(result)
-        result_one = _create_issue_search_results_json(
-            expected_results[:10], max_results=10, total=total
-        )
-        result_two = _create_issue_search_results_json(
-            expected_results[10:20], max_results=10, total=total
-        )
-        result_three = _create_issue_search_results_json(
-            expected_results[20:], max_results=6, total=total
-        )
+
+        if not max_results:
+            mocked_api_results = []
+            for i in range(start_at, total, batch_size):
+                mocked_api_result = _create_issue_search_results_json(
+                    expected_results[i : i + batch_size],
+                    max_results=batch_size,
+                    total=total,
+                )
+                mocked_api_results.append(mocked_api_result)
+        else:
+            mocked_api_results = [
+                _create_issue_search_results_json(
+                    expected_results[start_at : max_results + start_at],
+                    max_results=max_results,
+                    total=total,
+                )
+            ]
+
         mock_session = mock.Mock(name="mock_session")
         responses = mock.Mock(name="responses")
         responses.content = "_filler_"
-        responses.json.side_effect = [result_one, result_two, result_three]
+        responses.json.side_effect = mocked_api_results
         responses.status_code = 200
         mock_session.request.return_value = responses
         mock_session.get.return_value = responses
         self.jira._session.close()
         self.jira._session = mock_session
-        items = self.jira._fetch_pages(Issue, "issues", "search", 0, False, params)
-        self.assertEqual(len(items), total)
+        items = self.jira._fetch_pages(
+            Issue, "issues", "search", start_at, max_results, params=params
+        )
+
+        actual_calls = [[kall[1], kall[2]] for kall in self.jira._session.method_calls]
+        self.assertEqual(actual_calls, expected_calls)
+        self.assertEqual(len(items), total - start_at)
         self.assertEqual(
             {item.key for item in items},
-            {expected_r["key"] for expected_r in expected_results},
+            {expected_r["key"] for expected_r in expected_results[start_at:]},
         )
 
 
+@pytest.mark.parametrize(
+    "default_batch_sizes, item_type, expected",
+    [
+        ({Issue: 2}, Issue, 2),
+        ({Resource: 1}, Resource, 1),
+        (
+            {Resource: 1, Issue: None},
+            Issue,
+            None,
+        ),  # None -> backend determines batch-size
+        ({Resource: 1}, Dashboard, 1),  # fallback working correctly
+        ({}, Issue, 500),  # default value for Issue
+        ({}, Resource, 100),  # default value for everything else
+    ],
+)
```

Comment:
(nitpick, non-blocking) rather than using comments to specify conditions the preferred way is using the [pytest ids=](https://docs.pytest.org/en/6.2.x/example/parametrize.html#different-options-for-test-ids)

```suggestion
    ids=["modify Issue default", "modify Resource default", "let backend decide for Issue", "fallback", "default for Issue", "default value for everything else"],
)
```

the idea being it improves the readability of the test report, less of a counting game when a test fails

Reply:
Didn't know of the `ids`-arg. Nice!

Reply:
It is nice, I much prefer it compared to `parameterized.expand` way, where you can get away with creating an argument called "name" in the test function, or use a naming function, it feels a bit clunkier.